### PR TITLE
14 finalize read write of channel data

### PR DESCRIPTION
**Pull request by dominic22**
- differentiate between data which is public and which is masked when adding logs
- always return messageId for explorer
- always add created field
- possibility to sort data by created field

### CODE REVIEW

File: api/src/utils/channel-log-transformer/index.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export interface IPayload {
+	data?: any;
+	metadata?: any;
+	created?: any;
+	type?: string;
+}
+
+export class ChannelLogTransformer {
+	static transformStreamsMessages(data: StreamsMessage[]): ChannelData[] {
+		return data.map((data) => {
```
Comment (sobolev-alexey) [inline api/src/utils/channel-log-transformer/index.ts:13]: To make sure it won't fail I would add optional chaining to the `data` array inself and all keys of the `data` object

Another point - here is a name clash: you use `data` as iterable, and the object inside is also named `data`

File: api/src/utils/channel-log-transformer/index.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		});
+	}
+
+	static getChannelLog(publicPayload: IPayload, encryptedPayload: IPayload): ChannelLog {
+		return {
+			type: publicPayload.type,
+			metadata: publicPayload.metadata,
+			created: publicPayload.created,
+			payload: encryptedPayload.data,
+			publicPayload: publicPayload.data
```
Comment (sobolev-alexey) [inline api/src/utils/channel-log-transformer/index.ts:28]: Here and for the `getPayloads` function below I would also add optional chaining

File: api/src/database/channel-data.ts [hunk 3]
```diff
@@ -5,13 +5,20 @@ import { ChannelData } from '../models/types/channel-data';
...
+	identityId: string,
+	options: { limit?: number; index?: number; ascending: boolean }
+): Promise<ChannelData[]> => {
+	const { ascending, index, limit } = options;
+
 	const query = { channelAddress, identityId };
 	const skip = index > 0 ? index * limit : 0;
-	const options = limit != null ? { limit, skip, sort: { creationDate: 1 } } : undefined;
+	const sort = { 'channelLog.created': ascending ? 1 : -1 };
+	const opt = limit != null ? { limit, skip, sort } : { sort };
```
Comment (sobolev-alexey) [inline api/src/database/channel-data.ts:18]: `!=` or `!==` ?

Comment (dominic22) [reply api/src/database/channel-data.ts:18]: `!= null` is fine it also checks for undefined then

**Review by sobolev-alexey — APPROVED** [review]

**Pull request merged**